# netsurf: javascript
# 2 files, 125 changed lines
#
# Regenerate: tools/regen-patches.sh && tools/split-patches.py
#
diff -urN -x .git -x 'build-*' -x test -x frontends -x '*.o' -x '*.a' -x '*.d' -x '*.pc' -x build upstream/netsurf/content/handlers/javascript/fetcher.c browser/netsurf/content/handlers/javascript/fetcher.c
--- upstream/netsurf/content/handlers/javascript/fetcher.c	2026-09-20 23:29:00.073381249 +0100
+++ browser/netsurf/content/handlers/javascript/fetcher.c	2026-09-20 23:19:20.751100088 +0100
@@ -199,6 +199,7 @@
  *
  * \return NSERROR_OK on success or appropriate error code on faliure.
 */
+#ifndef __MACOS9__
 nserror fetch_javascript_register(void)
 {
 	lwc_string *scheme = lwc_string_ref(corestring_lwc_javascript);
@@ -215,3 +216,4 @@
 
 	return fetcher_add(scheme, &fetcher_ops);
 }
+#endif /* !__MACOS9__ */
diff -urN -x .git -x 'build-*' -x test -x frontends -x '*.o' -x '*.a' -x '*.d' -x '*.pc' -x build upstream/netsurf/content/handlers/javascript/js_content.c browser/netsurf/content/handlers/javascript/js_content.c
--- upstream/netsurf/content/handlers/javascript/js_content.c	1970-01-01 01:00:00.000000000 +0100
+++ browser/netsurf/content/handlers/javascript/js_content.c	2026-09-20 23:19:20.751530011 +0100
@@ -0,0 +1,123 @@
+/*
+ * Copyright 2012 Vincent Sanders <vince@kyllikki.org>
+ *
+ * This file is part of NetSurf, http://www.netsurf-browser.org/
+ *
+ * NetSurf is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; version 2 of the License.
+ *
+ * NetSurf is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+/**
+ * \file
+ * javascript content implementation
+ */
+
+#include <assert.h>
+#include <utils/string.h>
+#include <stdbool.h>
+#include <stdlib.h>
+
+#include "utils/errors.h"
+#include "utils/config.h"
+#include "content/content_protected.h"
+#include "content/content_factory.h"
+#include "content/hlcache.h"
+#include "utils/log.h"
+#include "utils/messages.h"
+#include "utils/utils.h"
+#include "netsurf/content.h"
+
+typedef struct javascript_content {
+	struct content base;
+} javascript_content;
+
+static nserror javascript_create(const content_handler *handler,
+		lwc_string *imime_type, const struct http_parameter *params,
+		llcache_handle *llcache, const char *fallback_charset,
+		bool quirks, struct content **c)
+{
+	javascript_content *script;
+	nserror error;
+
+	script = calloc(1, sizeof(javascript_content));
+	if (script == NULL)
+		return NSERROR_NOMEM;
+
+	error = content__init(&script->base, handler, imime_type, params,
+			llcache, fallback_charset, quirks);
+	if (error != NSERROR_OK) {
+		free(script);
+		return error;
+	}
+
+	*c = (struct content *) script;
+
+	return NSERROR_OK;
+}
+
+static bool javascript_convert(struct content *c)
+{
+	content_set_ready(c);
+	content_set_done(c);
+
+	return true;
+}
+
+static nserror 
+javascript_clone(const struct content *old, struct content **newc)
+{
+	javascript_content *script;
+	nserror error;
+
+	script = calloc(1, sizeof(javascript_content));
+	if (script == NULL)
+		return NSERROR_NOMEM;
+
+	error = content__clone(old, &script->base);
+	if (error != NSERROR_OK) {
+		content_destroy(&script->base);
+		return error;
+	}
+
+	*newc = (struct content *) script;
+
+	return NSERROR_OK;
+}
+
+static void javascript_destroy(struct content *c)
+{
+}
+
+static content_type javascript_content_type(void)
+{
+	return CONTENT_JS;
+}
+
+
+static const content_handler javascript_content_handler = {
+	.create = javascript_create,
+	.data_complete = javascript_convert,
+	.destroy = javascript_destroy,
+	.clone = javascript_clone,
+	.type = javascript_content_type,
+	.no_share = false,
+};
+
+static const char *javascript_types[] = {
+	"application/javascript", /* RFC 4329 */
+	"application/ecmascript", /* RFC 4329 */
+	"application/x-javascript", /* common usage */
+	"text/javascript", /* common usage */
+	"text/ecmascript", /* common usage */
+};
+
+CONTENT_FACTORY_REGISTER_TYPES(javascript, javascript_types, javascript_content_handler);
